tests/Encoder/PartnerizeS2SEncoderTest.php

```diff
@@ -12,15 +12,87 @@
 class PartnerizeS2SEncoderTest extends TestCase
 {
     /**
+     * @var PartnerizeS2SEncoder
+     */
+    private $encoder;
+
+    /**
+     * Setup class variables for testing
+     */
+    public function setUp(): void
+    {
+        $this->encoder = new PartnerizeS2SEncoder();
+    }
+
+    /**
+     * Clear class variables after testing
+     */
+    public function breakDown(): void
+    {
+        $this->encoder = null;
+    }
+
+    /**
+     * Test that everything properly encodes
+     *
      * @dataProvider encodingDataProvider
      */
     public function testEncoding($data, $expectedResult)
     {
-        $encoder = new PartnerizeS2SEncoder();
-        $result = $encoder->encode($data, PartnerizeS2SEncoder::FORMAT, []);
+        $result = $this->encoder->encode($data, PartnerizeS2SEncoder::FORMAT, []);
         $this->assertEquals($expectedResult, $result);
     }
 
+    /**
+     * Test that the support method only returns true on proper format
+     *
+     * @dataProvider formatDataProvider
+     */
+    public function testSupportsEncodingReturnsBoolean($encoding, $expected): void
+    {
+        $result = $this->encoder->supportsEncoding($encoding);
+        $this->assertEquals($expected, $result);
+    }
+
+    /**
+     * Test that the encoder throws a RuntimeException on bad input data
+     */
+    public function testRuntimeExceptionIfDataNotArrayOnEncoding(): void
+    {
+        $this->expectException(\RuntimeException::class);
+        $this->expectExceptionMessage('Expected first parameter to be an array, but got string.');
+
+        $this->encoder->encode('test', 'json');
+    }
+
+    /**
+     * Test that the encoder throws a RuntimeException on unnormalized data
+     */
+    public function testRuntimeExceptionIfValueNotScalarOnEncoding(): void
```

Comment:
Same here, what if you rename this test function to `testEncodeWillThrowRuntimeExceptionWhenDataValueIsNotScalar`? Then you'll read that you're actually testing the Encode function but will throw an exception when the given data value is not scalar.

Reply:
Refactored tests according to above feedback.